src/prommis/solid_handling/crusher.py

```diff
@@ -0,0 +1,216 @@
+
+"""
+
+from functools import partial
+from pyomo.environ import Var, log, Constraint, units as pyunits
+from pyomo.common.config import ConfigValue, ConfigDict, In
+from idaes.core import (
+    declare_process_block_class,
+    UnitModelBlockData,
+    useDefault,
+)
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.tables import create_stream_table_dataframe
+import idaes.logger as idaeslog
+
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("CrushAndBreakageUnit")
+class CrushAndBreakageUnitData(UnitModelBlockData):
+    CONFIG = (
+        ConfigDict()
+    )  # or  CONFIG = UnitModelBlockData.CONFIG() (not sure what's the difference)
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([False]),
+            default=False,
+            description="Dynamic model flag - must be False",
+            doc="""Crush unit is steady-state only""",
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be False",
+            doc="""Crush unit has no holdup.""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+**default** - useDefault.
+**Valid values:** {
+**useDefault** - use default package from parent model or flowsheet,
+**PropertyParameterObject** - a PropertyParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigDict(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigDict with arguments to be passed to a property block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}""",
+        ),
+    )
+
+    def build(self):
+        """
+        Begin building model (pre-DAE transformation).
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super().build()
+
+        self.properties_in = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            defined_state=True,
+            **self.config.property_package_args,
+        )
+        self.properties_out = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            defined_state=True,
+            **self.config.property_package_args,
+        )
+
+        tref = self.flowsheet().time.first()
+        statevars = self.properties_in[tref].define_state_vars()
+
+        for k, v in statevars.items():
```

Comment:
Add a comment here to indicate that this is writing a constraint of In=Out for all state variables which are not related to particle size